FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -89,18 +89,27 @@ - (instancetype)initWitAppOptions:(FIROptions *)appOptions appName:(NSString *)a
                                                       projectID:appOptions.projectID];
   return [self initWithAppOptions:appOptions
                           appName:appName
-        installationsIDController:IDController];
+        installationsIDController:IDController
+                prefetchAuthToken:YES];
 }
 
 /// The initializer is supposed to be used by tests to inject `installationsStore`.
 - (instancetype)initWithAppOptions:(FIROptions *)appOptions
                            appName:(NSString *)appName
-         installationsIDController:(FIRInstallationsIDController *)installationsIDController {
+         installationsIDController:(FIRInstallationsIDController *)installationsIDController
+                 prefetchAuthToken:(BOOL)prefetchAuthToken {
   self = [super init];
   if (self) {
     _appOptions = [appOptions copy];
     _appName = [appName copy];
     _installationsIDController = installationsIDController;
+
+    // Pre-fetch auth token.
+    if (prefetchAuthToken) {
+      [self authTokenWithCompletion:^(FIRInstallationsAuthTokenResult *_Nullable tokenResult,
```

Comment:
@ryanwilson Do you think it is fine to start the request from here? Or it is better to do it from `configureWithApp:` or maybe another place?

Reply:
Definitely not `configureWithApp:`,  perhaps this call should be moved to a different thread though?

Reply:
This method is async and non-blocking, so, I think we should be fine from as is from this perspective. We will check how FIS affects the startup time anyways, so I'll take a look what can be done if needed.

Reply:
SGTM.